src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1730,6 +1730,22 @@ bool checkValid(MethodSymbol candidate, bool isCandidate, DiagnosticBag specific
                         }
                     }
                 }
+                else if (LanguageVersion >= MessageID.IDS_FeatureAsyncMain.RequiredVersion() && taskEntryPoints.Count > 0 && viableEntryPoints.Count == 1)
+                {
+                    var taskCandidates = taskEntryPoints.Select(s => (Symbol)s.Candidate).ToImmutableArray();
+                    var taskLocations = taskCandidates.Select(s => s.Locations.First()).ToImmutableArray();
```

Comment:
>First [](start = 79, length = 5)

Please just use an array indexer.

Reply:
Same with all the `First`s below.

---
In reply to: [473197136](https://github.com/dotnet/roslyn/pull/46832#discussion_r473197136) [](ancestors = 473197136)

Reply:
```suggestion
                    var taskLocations = taskCandidates.SelectAsArray(s => s.Locations[0]);
```